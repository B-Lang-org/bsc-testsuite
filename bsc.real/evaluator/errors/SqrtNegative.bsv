import Real::*;

(* synthesize *)
module sysSqrtNegative();
    real y = sqrt(-4.0);

    rule r;
      $display("sqrt(-4.0) = " + realToString(y));
    endrule
endmodule