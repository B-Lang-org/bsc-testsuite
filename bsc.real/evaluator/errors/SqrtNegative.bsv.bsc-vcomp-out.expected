checking package dependencies
compiling SqrtNegative.bsv
code generation for sysSqrtNegative starts
Error: "SqrtNegative.bsv", line 5, column 14: (G0112)
  A square-root operation was applied to a negative number during elaboration.
  During elaboration of the body of rule `r' at "SqrtNegative.bsv", line 7,
  column 10.
  During elaboration of `sysSqrtNegative' at "SqrtNegative.bsv", line 4,
  column 8.
